app/lib/vita_partner_importer.rb

```diff
@@ -0,0 +1,61 @@
+require 'yaml'
+
+##
+# this module upserts Vita Partner data
+module VitaPartnerImporter
+  VITA_PARTNERS_YAML = Rails.root.join("db/vita_partners.yml")
+
+  ##
+  # output normally except in test
+  def say(message)
+    puts message unless Rails.env.test?
+  end
+
+  def upsert_vita_partners(yml = VITA_PARTNERS_YAML)
+    say "beginning partner upsert using environment: #{Rails.env}"
+
+    partners = YAML.load_file(yml)['vita_partners']
+    partners.each do |datum|
+      states = datum.delete('states')
+      sources = datum.delete('source_parameters')
+      partner = VitaPartner.find_by(zendesk_group_id: datum['zendesk_group_id'])
+      partner = partner.present? ? update_partner(partner, datum) : create_partner(datum)
+      refresh_partner_sources(partner, sources)
```

Comment:
bit confused about the difference between source_parameters and source_codes here...